### PR TITLE
Matched Impl and interface lifetimes

### PR DESCRIPTION
**Pull request by tchelidze**
Fix for #20

### CODE REVIEW

File: src/NServiceBus.Extensions.DependencyInjection/ServiceCollectionAdapter.cs [hunk 1]
```diff
@@ -73,13 +76,13 @@ public void RegisterSingleton<T>(T instance)
...
 
-        void RegisterInterfaces(Type component)
+        void RegisterInterfaces(Type component, ServiceLifetime lifetime)
         {
             var interfaces = component.GetInterfaces();
             foreach (var serviceType in interfaces)
             {
                 // see https://andrewlock.net/how-to-register-a-service-with-multiple-interfaces-for-in-asp-net-core-di/
-                serviceCollection.Add(new ServiceDescriptor(serviceType, sp => sp.GetService(component), ServiceLifetime.Transient));
+                serviceCollection.Add(new ServiceDescriptor(serviceType, sp => sp.GetService(component), lifetime));
```
Comment (timbussmann) [inline src/NServiceBus.Extensions.DependencyInjection/ServiceCollectionAdapter.cs]: this seems to be only necessary to be compatible with Castle.Windsor, but as long as the lifetimes of the forwarding registration and the "main" registration are equal, that shouldn't be an issue.

Comment (tchelidze) [reply src/NServiceBus.Extensions.DependencyInjection/ServiceCollectionAdapter.cs]: @timbussmann besides castle windsor i think there is one more issue which i tried to explain [here](https://github.com/Particular/NServiceBus.Extensions.DependencyInjection/issues/20#issuecomment-603252464)

**Comment by tchelidze** [comment]
> Do we have a similar bug in https://github.com/Particular/NServiceBus.Extensions.Hosting ?

[Looks like](https://github.com/Particular/NServiceBus.Extensions.Hosting/blob/master/src/NServiceBus.Extensions.Hosting/ServiceCollectionAdapter.cs#L82) yes

**Comment by tchelidze** [comment]
@andreasohlund @timbussmann any thought why checks are stuck?

**Comment by timbussmann** [comment]
@tchelidze for security reasons, external PRs aren't automatically built by the CI pipeline. We'll take care of that.

**Comment by timbussmann** [comment]
I'm merging this into a local branch which can then be merged into master via a separate PR.
Thanks for providing this PR @tchelidze 👍

**Pull request merged**